tmt/steps/provision/mrack.py

```diff
@@ -714,12 +714,14 @@ def _translate_tmt_hw(self, hw: tmt.hardware.Hardware) -> dict[str, Any]:
             """ Return hw requirements from given hw dictionary """
 
             assert hw.constraint
-
-            transformed = MrackHWAndGroup(
-                children=[
-                    constraint_to_beaker_filter(constraint, logger)
-                    for constraint in hw.constraint.variant()
-                    ])
+            if isinstance(hw.constraint, tmt.hardware.Or):
```

Comment:
Why is this needed when `constraint_to_beaker_filter()` seems to do the same thing?

```python
def constraint_to_beaker_filter(
        constraint: tmt.hardware.BaseConstraint,
        logger: tmt.log.Logger) -> MrackBaseHWElement:
    """ Convert a hardware constraint into a Mrack-compatible filter """

    if isinstance(constraint, tmt.hardware.And):
        return MrackHWAndGroup(
            children=[
                constraint_to_beaker_filter(child_constraint, logger)
                for child_constraint in constraint.constraints
                ]
            )

    if isinstance(constraint, tmt.hardware.Or):
        return MrackHWOrGroup(
            children=[
                constraint_to_beaker_filter(child_constraint, logger)
                for child_constraint in constraint.constraints
                ]
            )
```

IIUIC, when called, `constraint_to_beaker_filter()` runs `isinstance()` calls and handles `And` and `Or` when needed, so, to me, this is suspicious: `constraint_to_beaker_filter()` should take care of this, if it does not, then why, and how it needs to be fixed? Because adding a special bit of code here won't help, there may be nested `Or` uses, deeper in the tree of HW requirements, and this particular change would not handle those. That's why it's in `constraint_to_beaker_filter()`, it scans the whole tree and there shouldn't be a difference between the top-level one or a deeper one, i.e. this function is the one to debug and fix.